pandas/core/generic.py

```diff
@@ -233,6 +244,19 @@ def _init_mgr(self, mgr, axes=None, dtype=None, copy=False):
 
     # ----------------------------------------------------------------------
 
+    @property
+    def attrs(self) -> Dict[Hashable, Any]:
```

Comment:
Been a while since we've added something to a frame - do we have to make any consideration for dot access for frames where this was previously a valid label?

Reply:
Don't think so. We have a blanket statement that `.` is a convenience.

Reply:
FWIW, we don't need to expose this publicly yet. We can call it `._attrs` until we're ready.

My main motivation is ease of implementation in https://github.com/pandas-dev/pandas/pull/28394.

But xarray is an example of this being useful in practice.